spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageResourceBucket.java

```diff
@@ -0,0 +1,150 @@
+/*
+ *  Copyright 2017 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.storage;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+
+import com.google.cloud.storage.Bucket;
+import com.google.cloud.storage.BucketInfo;
+import com.google.cloud.storage.Storage;
+
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.WritableResource;
+import org.springframework.util.Assert;
+
+/**
+ * Implements {@link WritableResource} for buckets in Google Cloud Storage.
+ *
+ * @author Chengyuan Zhao
+ */
+public class GoogleStorageResourceBucket implements WritableResource {
+
+	private final Storage storage;
+
+	private final String bucketName;
+
+	private final boolean createBucketIfRequired;
```

Comment:
Where do we stand regarding automatic bucket creation?
@meltsufin suggested we shouldn't be doing it, and I'm tempted to agree.
I think users are expected to create the buckets on their own, even more so because bucket names are unique across GCS.

Reply:
I thought you were on board with the idea at the beginning? Tbh I think intentionally not giving this functionality  would be paternalism/parentalism. The API for creating buckets is super simple to use, and people could go wild with that with minimal effort as it is. Also, since we primarily let people use this functionality via @Value() annotations in source code, that's not conducive to wanton bucket creation anyway.

Reply:
I don't really have strong feelings on this matter.
You do have a point that there is an API for that already and we're just making it easy for users.

So, if @meltsufin is aboard, I'm OK with shipping this feature.